dotnet/Gherkin/Parser.cs

```diff
@@ -75,8 +75,20 @@ public enum RuleType
     [System.Runtime.CompilerServices.CompilerGeneratedAttribute()]
     public partial class Parser<T>
     {
+        public Parser()
+            : this(new AstBuilder<T>())
+        {
+        }
+
+        public Parser(AstBuilder<T> astBuilder)
+        {
+            this.AstBuilder = astBuilder;
+        }
+
         public bool StopAtFirstError { get; set;}
 
+        public AstBuilder<T> AstBuilder { get; }

```

Comment:
Given that the `astBuilder` is only used inside the `Parser` class, I wonder if it really needs a `get`, it seems to me that the `astBuilder` field could be declared like the [`dialectProvider` field in the TokenMatcher](https://github.com/cucumber/gherkin3/blob/master/dotnet/Gherkin/TokenMatcher.cs#L11).